modules/imgcodecs/src/grfmt_gdal.cpp

```diff
@@ -559,14 +559,18 @@ bool GdalDecoder::checkSignature( const String& signature )const{
 
 
     // look for NITF
-    std::string str = signature.c_str();
```

Comment:
This [should](https://github.com/opencv/opencv/blob/28db4a220741b33ba59a30fac6dd8e470f444cf6/modules/core/include/opencv2/core/cvstd.inl.hpp#L115-L119) work fine with null terminated symbols:
```
std::string str(signature);
```